### PR TITLE
JDK-8306696: Remove MetaspaceReclaimPolicy=aggressive and obsolete MetaspaceReclaimPolicy

### PR DESCRIPTION
**Pull request by tstuefe**
The diagnostic setting MetaspaceReclaimPolicy=aggressive is very rarely used, and I would like to remove it. This is a part of a larger effort to simplify metaspace coding and cut down on testing time, in preparation for upcoming changes to Metaspace with Lilliput.

MetaspaceReclaimPolicy had recently been demoted from an official to a diagnostic switch (see CSR [JDK-8302130](https://bugs.openjdk.org/browse/JDK-8302130)). See also the CSR text for a more in-depth explanation of what this switch does, its history, and why it could be removed for good.

The switch has not that much impact on RSS reduction but has the side effect of increasing VMA fragmentation compared to the default setting since it reduces the size of metaspace commit granules from 64K to 16K. The settings used by default (MetaspaceReclaimPolicy=balanced) are proven now in the field and can be used without alternatives (since any alternative has to be tested).

Since this would remove the last valid value for MetaspaceReclaimPolicy apart from its default "balanced" value, it makes sense to deprecate MetaspaceReclaimPolicy at the same time.

This allows us to cut down on Metaspace testing quite a bit since it removes one permutation from the test set.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8306696](https://bugs.openjdk.org/browse/JDK-8306696): Remove MetaspaceReclaimPolicy=aggressive and obsolete MetaspaceReclaimPolicy

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Roman Kennke](https://openjdk.org/census#rkennke) (@rkennke - **Reviewer**)
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13597/head:pull/13597` \
`$ git checkout pull/13597`

Update a local copy of the PR: \
`$ git checkout pull/13597` \
`$ git pull https://git.openjdk.org/jdk.git pull/13597/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13597`

View PR using the GUI difftool: \
`$ git pr show -t 13597`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13597.diff">https://git.openjdk.org/jdk/pull/13597.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13597#issuecomment-1518633020)

### CODE REVIEW

File: src/hotspot/share/runtime/arguments.cpp
Comment (dholmes-ora) [inline src/hotspot/share/runtime/arguments.cpp:507]: What you have done isn't deprecation, it is obsoletion - the flag is accepted but completely ignored (which is fine by me for a diagnostic flag). The above should be changed accordingly, or else continue to parse the flag and check for a valid value.

**Review by dholmes-ora — CHANGES_REQUESTED** [review]

**Comment by tstuefe** [comment]
```
thomas@starfish$ ./images/jdk/bin/java -XX:MetaspaceReclaimPolicy=balanced
OpenJDK 64-Bit Server VM warning: Option MetaspaceReclaimPolicy was deprecated in version 21.0 and will likely be removed in a future release.
```

File: src/hotspot/share/runtime/arguments.cpp
Comment (dholmes-ora) [reply src/hotspot/share/runtime/arguments.cpp:507]: If the flag only has one valid value now then it is effectively obsolete, so you may as well mark it as such. Existing users of the flag will get a warning either way.

Comment (tstuefe) [reply src/hotspot/share/runtime/arguments.cpp]: Okay, done so.

**Review by coleenp — APPROVED** [review]
Looks good.

**Comment by tstuefe** [comment]
> Looks good.

Thanks @coleenp !

> Looks ok to me. 

Thanks @rkennke !

>If demoting the flag to diagnostic warrants a CSR, does it also warrant a CSR to remove it altogether? As a courtesy

Let's not... it is a diagnostic flag now, and the reason I made it so was to be free to do stuff with it without going through CSR.

I'll wait for the tests to finish and for Davids approval.

**Comment by tstuefe** [comment]
Update: changed issue text from "deprecate" to "obsolete"

**Comment by tstuefe** [comment]
@dholmes-ora are you okay with this last version?

**Review by dholmes-ora — APPROVED** [review]
That seems fine. Thanks.

**Comment by tstuefe** [comment]
Thank you David!

/integrate

**Comment by DamonFool** [comment]
Hi @tstuefe ,
It seems two tests are missed here: https://github.com/openjdk/jdk/pull/13714
Please have a look.
Thanks.

**Comment by tstuefe** [comment]
@DamonFool  @dholmes-ora 

Sorry for that. I could have sworn all Metaspace tests run under tier1 and hence as part of the GHAs, that's why I felt safe to integrate. But I forgot we had removed most of these test from tier1 with JDK-8272846.

**Pull request closed without merging**
